pkg/karmadactl/join.go

```diff
@@ -277,32 +310,19 @@ func JoinCluster(controlPlaneRestConfig, clusterConfig *rest.Config, clusterName
 		url, err := clusterConfig.Proxy(nil)
 		if err != nil {
 			klog.Errorf("clusterConfig.Proxy error, %v", err)
-			return err
+			return nil, err
 		}
 		clusterObj.Spec.ProxyURL = url.String()
 	}
 
+	controlPlaneKarmadaClient := karmadaclientset.NewForConfigOrDie(controlPlaneConfig)
 	cluster, err := CreateClusterObject(controlPlaneKarmadaClient, clusterObj, false)
 	if err != nil {
 		klog.Errorf("failed to create cluster object. cluster name: %s, error: %v", clusterName, err)
-		return err
-	}
-
-	patchSecretBody := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			OwnerReferences: []metav1.OwnerReference{
-				*metav1.NewControllerRef(cluster, clusterResourceKind),
-			},
-		},
-	}
-
-	err = util.PatchSecret(controlPlaneKubeClient, secretNamespace, secretName, types.MergePatchType, patchSecretBody)
-	if err != nil {
-		klog.Errorf("failed to patch secret %s/%s, error: %v", secret.Namespace, secret.Name, err)
-		return err
```

Comment:
I can't remember why we need to patch the owner reference. 
Why not add the reference when creating the secrets in #67 ?. @mrlihanbo